templates/geosafe/analysis/dynamic_scripts/create/transparency_control.html

```diff
@@ -152,6 +152,31 @@
                 $("#impact-legend-img").attr("src", legend_url);
                 impact_layer.setZIndex(3);
             }
+
+            // Zooming magic
+            // If only one of the 3 layers is selected then zoom to the layer
+            if ((is_hazard != is_exposure) != is_impact) {
+              map.fitBounds([[layer_bbox_y1, layer_bbox_x1], [layer_bbox_y0, layer_bbox_x0]]);
+            }
+            // If hazard and exposure selected intersect both extents
+            if ((layer_purpose=='hazard' && is_exposure) || (layer_purpose=='exposure' && is_hazard)) {
+              // Intersect current extent to the input layer's extent
+              var new_sw = [
+                  Math.max(
+                      current_bounds.getSouthWest().lat,
+                      layer_bounds.getSouthWest().lat),
+                  Math.max(
+                      current_bounds.getSouthWest().lng,
+                      layer_bounds.getSouthWest().lng)];
+              var new_ne = [
+                  Math.min(
+                      current_bounds.getNorthEast().lat,
+                      layer_bounds.getNorthEast().lat),
+                  Math.min(
+                      current_bounds.getNorthEast().lng,
+                      layer_bounds.getNorthEast().lng)];
+              map.fitBounds([new_sw, new_ne]);
+            }
```

Comment:
@akbargumbira , CMIIW, this algorithm will not work if current bounds and layer_bounds doesn't intersects. But since user can only click exposure that is in the current bounds, so I think this is ok.
But, if you think it is quick to implement (or if you have any good js library to suggest), can you handle international dateline too?
If this will take too much time, I think we can skip that.
CC @gubuntu , what do you think?
Other option is too let the server handle the extent calculation using django.contrib.geos, perhaps.
Otherwise, LGTM.